php/libraries/LorisForm.class.inc

```diff
@@ -1884,20 +1884,24 @@ class LorisForm
                  */
                 $el = $this->createBase($args[1], $args[2], $args[4]);
 
-                $el['type']        = $type;
+                $el['type']        = 'advcheckbox';
                 $el['checkStates'] = $args[5];
                 $el['_text']       = $args[3];
             } else {
                 $el         = $this->createBase($elname, $label, $options);
-                $el['type'] = $type;
+                $el['type'] = 'advcheckbox';
             }
             break;
+        case 'date':
+            $el            = $this->createBase($elname, $label, $attribs);
+            $el['type']    = 'date';
+            $el['options'] = $options;
+            break;
         case 'time':
         case 'textarea':
-        case 'date':
         case 'password':
             $el         = $this->createBase($elname, $label, $options);
-            $el['type'] = $type;
+            $el['type'] = 'password';
```

Comment:
this is wrong in the time/textarea cases which fallthrough to here.

(In my initial comment I was just referring to the 'date' case that you were modifying.)